doc/devel/documenting_mpl.rst

```diff
@@ -181,135 +178,81 @@ Formatting
 
-* Please familiarize yourself with the Sphinx directives for `inline
-  markup`_. Matplotlib's documentation makes heavy use of cross-referencing and
-  other semantic markup. For example, when referring to external files, use the
-  ``:file:`` directive.
+* Matplotlib's documentation makes heavy use of cross-referencing and
+  other semantic markup. Details can be found in Sphinx directives for `inline
+  markup`_. For example, when referring to external files, the
+  ``:file:`` directive should be used.
 
-* Function arguments and keywords should be referred to using the *emphasis*
-  role. This will keep matplotlib's documentation consistent with Python's
-  documentation::
+* Please keep the :ref:`glossary` in mind when writing documentation. You can
+  create a references to a term in the glossary with the ``:term:`` role.
+
+* The autodoc extension will handle index entries for the API, but additional
+  entries in the index_ need to be explicitly added.
+
+* Please limit the text width of docstrings to 79 characters.
+
+Function arguments
+------------------
+Function arguments and keywords should be referred to using the *emphasis*
+role. This will keep Matplotlib's documentation consistent with Python's
+documentation::
 
     Here is a description of *argument*
 
-  Please do not use the `default role`::
+Please do not use the `default role`::
 
     Please do not describe `argument` like this.
 
-  nor the ``literal`` role::
+nor the ``literal`` role::
 
     Please do not describe ``argument`` like this.
 
 * Sphinx does not support tables with column- or row-spanning cells for
-  latex output. Such tables can not be used when documenting matplotlib.
+  latex output. Such tables can not be used when documenting Matplotlib.
 
-* Mathematical expressions can be rendered as png images in html, and in the
-  usual way by latex. For example:
+Maths
+-----
+Mathematical expressions can be rendered as png images in html, and in the
+usual way by latex. For example:
 
   ``:math:`\sin(x_n^2)``` yields: :math:`\sin(x_n^2)`, and::
 
     .. math::
 
       \int_{-\infty}^{\infty}\frac{e^{i\phi}}{1+x^2\frac{e^{i\phi}}{1+x^2}}
 
-  yields:
+yields:
 
   .. math::
 
     \int_{-\infty}^{\infty}\frac{e^{i\phi}}{1+x^2\frac{e^{i\phi}}{1+x^2}}
 
-* Interactive IPython sessions can be illustrated in the documentation using
-  the following directive::
+IPython code
+------------
+Interactive IPython sessions can be illustrated in the documentation using
+the following directive::
 
     .. sourcecode:: ipython
 
       In [69]: lines = plot([1,2,3])
 
-  which would yield:
+which would yield:
 
   .. sourcecode:: ipython
 
     In [69]: lines = plot([1,2,3])
 
-* Footnotes [#]_ can be added using ``[#]_``, followed later by::
+Footnotes
+---------
+Footnotes [#]_ can be added using ``[#]_``, followed later by::
 
     .. rubric:: Footnotes
 
     .. [#]
 
-  .. rubric:: Footnotes
-
```

Comment:
I've been fairly ruthless deleting stuff here, as how to do all this properly is documented in the numpydoc guide (which I will add a link to in this document at some point)